src/helpers/pivot/table_spreadsheet_pivot.ts

```diff
@@ -174,10 +191,13 @@ export class SpreadsheetPivotTable {
         pivotWidth -= this.measures.length;
       }
       const domainArray: PivotTableCell[][] = [];
-      const startRow = displayColumnHeaders ? 0 : this.columns.length;
+      const skippedRows = this.getSkippedRows(visibilityOptions);
       for (let col = 0; col < pivotWidth; col++) {
         domainArray.push([]);
-        for (let row = startRow; row < pivotHeight; row++) {
+        for (let row = 0; row < pivotHeight; row++) {
+          if (skippedRows.has(row)) {
+            continue;
+          }
           if (!displayTotals && row === pivotHeight) {
```

Comment:
this `if` should be moved to `getSkippedRows`

Reply:
Actually it should be removed, the condition will never be true (for loop stopped at `row = pivotHeight - 1`